### PR TITLE
[media-gfx/blender] Update minimum LLVM version of deps and python deps clean up

### PR DESCRIPTION
**Pull request by DarkDefender**
@juippis I updated the more recent ebuilds with a fix for the Cython implicit runtime dep that we discussed previously.

I removed the ability to build and bundle system python libraries as at least I don't think this feature brings much to the table within Gentoo portage. We already have very good tools to ensure that the correct python version and libraries are installed on the system, so I don't think we need to expose this functionality.

### CODE REVIEW

**Comment by DarkDefender** [comment]
@thesamesam seems like the CI bot got stuck.
Is there anyway to kick it?

**Comment by juippis** [comment]
I'm gonna paste some parts I've discovered:
```
>>> Install media-gfx/blender-3.0.1-r5 into /var/tmp/portage/media-gfx/blender-3.0.1-r5/image
!!! dobin: /var/tmp/portage/media-gfx/blender-3.0.1-r5/work/blender-3.0.1_build/bin/cycles does not    exist
 * ERROR: media-gfx/blender-3.0.1-r5::gentoo failed (install phase):
 *   dobin failed
```
(this is also on -9999)

```
/usr/lib/gcc/x86_64-pc-linux-gnu/11.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libcycles_graph.   a(node_xml.cpp.o): undefined reference to symbol '_ZNK4pugi13xml_attributentEv'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libpugixml. so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
...
 * ERROR: media-gfx/blender-2.93.9::gentoo failed (compile phase):
```

Blender:
```
 * QA Notice: The ebuild is installing to one or more unexpected paths:
 *
 *   /usr/cycles
```

```
^[[33m * ^[[39;49;00mQA Notice: This package installs one or more .desktop files that do not
^[[33m * ^[[39;49;00mpass validation.
^[[33m * ^[[39;49;00m
^[[33m * ^[[39;49;00m   /usr/share/applications/blender-3.1.desktop: warning: key "X-KDE-              RunOnDiscreteGpu" in group "Desktop Entry" is deprecated
^[[33m * ^[[39;49;00m
```
(in -9999 too)

Then 
```
/var/tmp/portage/vbslogs/build/dev-lang/ispc-9999:20220517-170138.log: * ERROR: dev-lang/ispc-9999::gentoo failed (test phase):
/var/tmp/portage/vbslogs/build/dev-lang/ispc-1.17.0-r1:20220517-184118.log: * ERROR: dev-lang/ispc-1.17.0-r1::gentoo failed (test phase):
```
```
/var/tmp/portage/vbslogs/build/media-gfx/blender-2.93.9:20220517-195705.log: * ERROR: media-gfx/blender-2.93.9::gentoo failed (test phase):
/var/tmp/portage/vbslogs/build/media-gfx/blender-3.1.2:20220517-182235.log: * ERROR: media-gfx/blender-3.1.2::gentoo failed (test phase):
```

I can run blender (2.93, 3.0, 3.3 - 3.1 doesn't start) so I'm happy, but do figure out the "cycles" issues and I'll approve.

**Comment by DarkDefender** [comment]
Thanks for the feedback!
I talked to the main Cycles developer (Brecht). It seems like the "standalone" option is not meant to be used for non development purposes. At least his intention with it is to make it easier to developer features inside of cycles and test them without having to touch any of the Blender code. It is also used as an example of how to use the cycles API.

IE it is an example project that is more of a "this is how you can create a cycles standalone render if you want to use it in your own program" kinda thing. It is not meant to be a binary non developers should use.

Therefore I decided to remove this option from the newer ebuilds.

> Notice: This package installs one or more .desktop files that do not pass validation.

This is a known issue where Blender uses deprecated flags to signal that the program should run on the most powerful GPU in the system. The tricky part here is that I haven't found any concrete info of when this flag will be dropped and which DE version support the new flags. 

Because of this I can give the other Blender developers any concrete data when it will be safe to drop these.
Do you have any clue?

> Linking errors

Which pugixml version are you using?

> Blender test failures

It is expected that there are some test failures on older Blender versions.
The live ebuild should pass all tests however.

> Ispc test failures

I can't reproduce these test fails on my end.
Which tests are failing?

**Review by juippis — APPROVED** [review]
Heh, 
```
The following tests FAILED:
	 91 - cycles_hair_cpu (Failed)
Errors while running CTest
Output from these tests are in: /var/tmp/portage/media-gfx/blender-9999/work/blender-9999_build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
 * ERROR: media-gfx/blender-9999::gentoo failed (test phase):
```
```
The following tests FAILED:
	106 - compositor_filter_test (Failed)
Errors while running CTest
Output from these tests are in: /var/tmp/portage/media-gfx/blender-2.93.9/work/blender-2.93.9_build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
 * ERROR: media-gfx/blender-2.93.9::gentoo failed (test phase):
```
```
 * Messages for package media-gfx/blender-2.93.9:
 * SED: the following did not cause any changes
 *     sed -e "s|GENERATE_HTMLHELP      = YES|GENERATE_HTMLHELP      = NO|" -i doc/doxygen/Doxyfile || die;
 * please use append-cppflags for preprocessor flags
```
```
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
Compilation of test ./tests/launch-5.ispc failed
/usr/bin/x86_64-pc-linux-gnu-ld: ./tests/launch-5.ispc.o: relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/x86_64-pc-linux-gnu-ld: failed to set dynamic section sizes: bad value
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
Compilation of test ./tests/launch-9.ispc failed
/usr/bin/x86_64-pc-linux-gnu-ld: ./tests/launch-9.ispc.o: relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/x86_64-pc-linux-gnu-ld: failed to set dynamic section sizes: bad value
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
Compilation of test ./tests/idiv-uint16.ispc failed
/usr/bin/x86_64-pc-linux-gnu-ld: ./tests/idiv-uint16.ispc.o: relocation R_X86_64_32S against `.      rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/x86_64-pc-linux-gnu-ld: failed to set dynamic section sizes: bad value
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
Compilation of test ./tests/gs-double-improve-progindex-plus-unif.ispc failed
/usr/bin/x86_64-pc-linux-gnu-ld: ./tests/gs-double-improve-progindex-plus-unif.ispc.o: relocation    R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/x86_64-pc-linux-gnu-ld: failed to set dynamic section sizes: bad value
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
Compilation of test ./tests/print_varying-i8.ispc failed
/usr/bin/x86_64-pc-linux-gnu-ld: ./tests/print_varying-i8.ispc.o: relocation R_X86_64_32 against `.  rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/x86_64-pc-linux-gnu-ld: failed to set dynamic section sizes: bad value
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
Compilation of test ./tests/print_varying-i64.ispc failed
/usr/bin/x86_64-pc-linux-gnu-ld: ./tests/print_varying-i64.ispc.o: relocation R_X86_64_32 against `. rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/x86_64-pc-linux-gnu-ld: failed to set dynamic section sizes: bad value
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
Compilation of test ./tests/print_uniform-i8.ispc failed
/usr/bin/x86_64-pc-linux-gnu-ld: ./tests/print_uniform-i8.ispc.o: relocation R_X86_64_32 against `.  rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/x86_64-pc-linux-gnu-ld: failed to set dynamic section sizes: bad value
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
 Done 1585 / 1585 [./tests_errors/soa-2.ispc]
...
...
 * ERROR: dev-lang/ispc-9999::gentoo failed (test phase):
 *   Testing failed under python3.10
 *
 * Call stack:
 *     ebuild.sh, line 127:  Called src_test
 *   environment, line 3738:  Called die
 * The specific snippet of code:
 *       PATH="${BUILD_DIR}/bin:${PATH}" ${EPYTHON} ./run_tests.py || die "Testing failed under      ${EPYTHON}"
```

But overall everything **works**. Thanks a lot!

File: media-libs/osl/osl-1.11.17.0-r3.ebuild [hunk 1]
```diff
@@ -6,7 +6,7 @@ EAPI=7
 PYTHON_COMPAT=( python3_{8,9,10} )
 
 # check this on updates
-LLVM_MAX_SLOT=13
+LLVM_MAX_SLOT=14
```
Comment (thesamesam) [inline media-libs/osl/osl-1.11.17.0-r3.ebuild:9]: Need to revbump for these, fwiw (it's not the same as py compat which changes use flags)

File: dev-lang/ispc/ispc-1.17.0-r1.ebuild [hunk 2]
```diff
@@ -4,7 +4,7 @@
 EAPI=7
 
 PYTHON_COMPAT=( python3_{8..10} )
-LLVM_MAX_SLOT=13
+LLVM_MAX_SLOT=14
```
Comment (thesamesam) [inline dev-lang/ispc/ispc-1.17.0-r1.ebuild:7]: And these

Comment (juippis) [reply dev-lang/ispc/ispc-1.17.0-r1.ebuild:7]: Oh indeed, these are in RDEPEND.

**Pull request closed without merging**